nixos/modules/services/misc/mollysocket.nix

```
@@ -0,0 +1,113 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mollysocket;
+  configuration = format.generate "mollysocket.conf" cfg.settings;
+  format = pkgs.formats.toml { };
+  package = pkgs.writeShellScriptBin "mollysocket" ''
+    exec ${lib.getExe pkgs.mollysocket} -c ${configurarion} "$@"
```

small nitpick:

```suggestion
    export MOLLY_CONF=${configuration}
    exec ${lib.getExe pkgs.mollysocket} "$@"
```

would allow users of the cli to still provide their own config file via `-c`, if they ever want to use a different one from the one the module generated.

Currently, giving `-c` to the wrapper would result in `error: the argument '--config <FILE>' cannot be used multiple times`. But if the config file's location is given via environment variable instead, [supplying a different location via `-c` will take precedence](https://github.com/mollyim/mollysocket?tab=readme-ov-file#configuration) & work as expected.